cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/AbstractConfigurer.java

```diff
@@ -62,12 +62,19 @@ protected AbstractConfigurer(Id.Namespace deployNamespace, Id.Artifact artifactI
     return plugins;
   }
 
-  protected void addPlugins(Map<String, Plugin> plugins) {
-    Set<String> duplicatePlugins = Sets.intersection(plugins.keySet(), getPlugins().keySet());
-    Preconditions.checkArgument(duplicatePlugins.isEmpty(),
+  protected void addPlugins(Map<String, Plugin> pluginsToAdd) {
+    Map<String, Plugin> existingPlugins = getPlugins();
+    // We don't allow adding different plugin with same id. Adding same plugin with to an id is allowed as it just
+    // means that the plugin is being registered in the configurer again and that is fine. We do this for
+    // DataPipelineApp. DataPipelineApp.configure() method generates spec through BatchPipelineSpecGenerator which
```

Comment:
Same here, the message should be generic.

Reply:
Done.